### ISSUE TITLE
is_number doesn't handle negative values

### ISSUE THREAD

**Issue by demalexx** [nithinmurali/pygsheets#470]
**Describe the bug**
In 2.0.4 `is_number` was changed in such a way it doesn't recognize negative float as a number:
```
>>> from pygsheets.utils import is_number
>>> is_number(-1.23)
False
```

Due to this `pygsheets.cell.Cell.get_json` falls down the conditions and reaches the line 
```
elif type(self._value) is str or type(self._value) is unicode:
```
which crashes:
```
NameError: name 'unicode' is not defined
```

I use python 3.8.

**To Reproduce**
```
sheet.update_cells([[Cell('A1', -1.23]])
...
NameError: name 'unicode' is not defined
...
```

**System Information**
 - OS: Ubuntu 20.04.1 LTS
 - pygsheets version: 2.0.4
 - pygsheets installed from (github or pypi): pypi

**Comment by nithinmurali** [comment]
ahh, the fix to this #425 introduced this bug. we need some more tets and a fool proof method to check is_number. 

maybe we should revert to 

```
def is_number(n):
    try:
        float(n)
    except ValueError:
        return False
    return True
```

and add `'_' not in n` .

**Comment by nithinmurali** [comment]
this looks critical. One its finalized i will make a patch release.
